File: server/src/main/java/server/objects/Album.java
package server.objects;

/**
 * Class representing an album.
 */
public class Album {
    private final String albumName, description, authorName, albumTime;
    private final long albumId;

    public Album(long albumId, String albumName, String authorName, String description, String albumTime) {
        this.albumName = albumName;
        this.description = description;
        this.authorName = authorName;
        this.albumId = albumId;
        this.albumTime = albumTime;
    }

    /**
     * @return the album's description
     */
    public String getDescription() {
        return description;
    }

    /**
     * @return the album's name
     */
    public String getAlbumName() {
        return albumName;
    }

    /**
     * @return the timestamp the album was created at
     */
    public String getAlbumTime() {
        return albumTime;
    }

    /**
     * @return the album's id
     */
    public long getAlbumId() {
        return albumId;
    }

    /**
     * @return the author of the album's name
     */
    public String getAuthorName() {
        return authorName;
    }
}

File: web_client/js/models/Album.js
(function () {

    class Album {
        constructor(id, name, description, author, date) {
            this.id = id
            this.name = name
            this.description = description
            this.author = author
            this.date = date
        }

        static fromJson(data) {

            if (typeof data !== "object") {
                data = JSON.parse(data)
            }

            return new Album(data.albumId, data.albumName, data.description, data.authorName, data.albumTime)
        }

        toJson() {
            return JSON.stringify({
                albumId: this.id,
                albumName: this.name,
                description: this.description,
                authorName: this.author,
                albumTime: this.date,
            })
        }
    }


    window.Models.Album = Album
})()